### PR TITLE
READY: Decouple express

### PR DESCRIPTION
**Pull request by tandrewnichols**
This PR removes references to req and res to make ftoggle a standalone module not dependent on the express interface. It also updates the syntax to ES6+.

### CODE REVIEW

File: lib/user-config.js [hunk 1]
```diff
@@ -1,20 +1,16 @@
...
-  var winner = null;
+const pickWinner = function(features) {
+  let floor = 0;
+  let rand = math.random();
+  let winner = null;
 
-  for (var name in features) {
-    var feature = features[name];
-    var traffic = getTraffic(feature, bot);
+  for (let name in features) {
```
Comment (scottkipfer) [inline lib/user-config.js]: what is features structure like?
could you simplify this with:
```
for (traffic of features) {
....
}
```

Comment (tandrewnichols) [reply lib/user-config.js]: Features is an object.

Comment (tandrewnichols) [reply lib/user-config.js]: For/of is for iterables . . . I think that doesn't work with objects right?

Comment (tandrewnichols) [reply lib/user-config.js]: I _don't_ need the `feature` intermediate variable though.

**Comment by scottkipfer** [comment]
👍

File: lib/settings-builder.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+const _ = require('lodash');
+
+const build = exports.build = function(toggles, featureConfig) {
+  if (!toggles || !toggles.e) {
+    return {};
+  }
+
+  let settings = featureConfig && featureConfig.settings ? _.clone(featureConfig.settings) : {};
```
Comment (scottkipfer) [inline lib/settings-builder.js]: es6 opportunity... maybe?
``` let settings = featureConfig && featureConfig.settings ? {...featureConfig.settings} " {}; ```

Comment (tandrewnichols) [reply lib/settings-builder.js]: sure

File: lib/ftoggle.js [hunk 3]
```diff
@@ -22,60 +23,60 @@ class Ftoggle {
...
 
-  getFeatureVals() {
-    return this.featureVals;
+  getSettings() {
+    return this.settings;
   }
 
-  getConfigForFeature(feature) {
-    return _.get(this.toggleConfig, `${feature}.conf`);
+  getSettingsForFeature(feature) {
```
Comment (scottkipfer) [inline lib/ftoggle.js]: is there ever a chance of feature being `null` or `undefined`?  what would happen for
```
`${feature}.settings`
```

would it just be ... ` null.settings`

Comment (tandrewnichols) [reply lib/ftoggle.js]: I think it might actually blow up, but we pretty much only call this internally.

**Comment by scottkipfer** [comment]
👍  again

**Pull request merged**